config/routes.rb

```diff
@@ -38,9 +38,11 @@
       resources :totp_authentications, only: %i(create)
       resources :promo_registrations, only: %i(index create)
     end
-    resources :site_banners, only: %i(new create), controller: 'publishers/site_banners' do
+    resources :site_banners, controller: 'publishers/site_banners' do
       collection do
         get :fetch
+        get :channels
```

Comment:
So I think our controller method could be alternatively named `index` because that is by default included as a resource on `site_banner` currently. Then we don't have to define this `get :channels`

Reply:
That's true, I think I'm more of a fan of explicit definitions though and `channels` gives a better idea of what the function does than `index` IMO